packages/jellyfish-api-core/__tests__/category/loan/getCollateralToken.test.ts

```diff
@@ -0,0 +1,273 @@
+      token: 'AAPL',
+      factor: new BigNumber(0.5),
+      priceFeedId
+    }]) // Activate at next block
+    await testing.generate(1)
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should getCollateralToken', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({
+      height: await testing.container.getBlockCount()
+    })
+    expect(data).toStrictEqual({
+      [collateralTokenId]: {
+        token: 'AAPL',
+        factor: new BigNumber(0.5),
+        priceFeedId,
+        activateAfterBlock: new BigNumber(await testing.container.getBlockCount())
+      }
+    })
+  })
+
+  it('should getCollateralToken with empty string if height is below current height', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({ height: 50 })
+    expect(data).toStrictEqual({})
+  })
+
+  it('should getCollateralToken if height is after current height', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({ height: 150 })
+    expect(data).toStrictEqual({
+      [collateralTokenId]: {
+        token: 'AAPL',
+        factor: new BigNumber(0.5),
+        priceFeedId,
+        activateAfterBlock: new BigNumber(await testing.container.getBlockCount())
+      }
+    })
+  })
+})
+
+describe('Loan getCollateralToken with no parameter', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should getCollateralToken', async () => {
+    await testing.token.create({ symbol: 'AAPL' })
+    await testing.generate(1)
+
+    const priceFeedId1 = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'AAPL',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const collateralTokenId1 = await testing.container.call('setcollateraltoken', [{
+      token: 'AAPL',
+      factor: new BigNumber(0.5),
+      priceFeedId: priceFeedId1
+    }]) // Activate at next block
+    await testing.generate(1)
+    const blockCount = new BigNumber(await testing.container.getBlockCount())
+
+    await testing.token.create({ symbol: 'TSLA' })
+    await testing.generate(1)
+
+    const priceFeedId2 = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'TSLA',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const collateralTokenId2 = await testing.container.call('setcollateraltoken', [{
+      token: 'TSLA',
+      factor: new BigNumber(1),
+      priceFeedId: priceFeedId2,
+      activateAfterBlock: 120
+    }])
+    await testing.generate(1)
+
+    // Only remain AAPL which has already been activated
+    {
+      const data = await testing.rpc.loan.getCollateralToken({})
```

Comment:
if you call getCollateralToken without parameters, it return a list of all activated token ?
Seems weird, it should always return a single token and list should accept a height parameter or activated boolean.

Reply:
getcollateraltoken will return all current collateral tokens. If there is none then it will be empty list. listcollateraltokens will return all txs over time

Reply:
Well hahahahaha

Reply:
@fuxingloh

Reply:
if you call getCollateralToken without parameters, it return a list of all activated token ? => Actually, the best is C++ programmer create another rpc for this, maybe called listCurrentCollateralTokens and for listCollateralTokens, change it to listAllCollateralTokens

Reply:
```suggestion
      const data = await testing.rpc.loan.getCollateralToken()
```

Reply:
I adopted your suggestion. And changed `collateralToken: GetCollateralToken` to `collateralToken?: GetCollateralToken` to fix the compilation error raised from that 

<img width="1030" alt="Screenshot 2021-09-10 at 11 16 33 AM" src="https://user-images.githubusercontent.com/7101806/132793928-e66e8f88-5b7e-4832-afa7-d390992b3ba4.png">

Reply:
Test run failed. To fix the test, I change the code to  
<img width="897" alt="Screenshot 2021-09-10 at 11 27 31 AM" src="https://user-images.githubusercontent.com/7101806/132794828-abd65860-ce7d-4f1f-9ac4-70419fca751d.png">